src/utilities.jl

```diff
@@ -1,6 +1,28 @@
+# Utility function for getting chunks of an ND-array along a particular dimension
+chunk(A, k::Int; dim::Int = 1) = 
+    (selectdim(A, dim, i) for i in Iterators.partition(axes(A,dim), cld(size(A,dim), k)));
```

Comment:
It might be worth providing a gradient for this, to make one array like A, not 3.

Reply:
Also, if we are keeping this in Metalhead.jl, then it should be renamed so it doesn't conflict with `Flux.chunk`.

Reply:
Made it `chunks` for the time being, until https://github.com/JuliaML/MLUtils.jl/pull/47 goes through and we can change to using that instead